src/context/AppContext.tsx

```diff
@@ -0,0 +1,223 @@
+    "bannerHome",
+    0,
+  );
+  const homeBannerEl = useRef<HTMLElement>(null);
+  const [isBurgerOpen, setIsBurgerOpen] = useState<boolean>(false);
+  const [phonesList, setPhonesList] = useState<Product[]>([]);
+  const [tabletsList, setTabletsList] = useState<Product[]>([]);
+  const [accessoriesList, setAccessoriesList] = useState<Product[]>([]);
+  const [productsList, setProductsList] = useState<Product[]>([]);
+  const [favorite] = useState<number>(0);
+
+  const categories: Category[] = [
+    {
+      id: "mobile-phones",
+      array: phonesList,
+      name: "Mobile phones",
+      img: "/react_phone-catalog/img/sectionNames/category-phones.png",
+      backgroundColor: "#6d6474",
+      to: "/phones",
+    },
+    {
+      id: "tablets",
+      array: tabletsList,
+      name: "Tablets",
+      img: "/react_phone-catalog/img/sectionNames/category-tablets.png",
+      backgroundColor: "#dd8d92",
+      to: "/tablets",
+    },
+    {
+      id: "accessories",
+      array: accessoriesList,
+      name: "Accessories",
+      img: "/react_phone-catalog/img/sectionNames/category-accessories.png",
+      backgroundColor: "#973d5f",
+      to: "/accessories",
+    },
+  ];
+
+  useEffect(() => {
+    const fetchProducts = async () => {
+      try {
+        const response = await fetch("/react_phone-catalog/api/products.json");
+
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
+
+        const data: Product[] = await response.json();
+
+        const spitedCategories = data.reduce(
+          (
+            acc: Record<Product["category"], Product[]>,
+            currentProduct: Product,
+          ) => {
+            const key = currentProduct.category;
+
+            acc[key] = acc[key]
+              ? [...acc[key], currentProduct]
+              : [currentProduct];
+            return acc;
+          },
+          {},
+        );
+
+        setProductsList(data);
+        setPhonesList(spitedCategories.phones);
+        setTabletsList(spitedCategories.tablets);
+        setAccessoriesList(spitedCategories.accessories);
+      } catch (err) {
+        console.error("Something went wrong", err);
+      }
+    };
+
+    fetchProducts();
+  }, []);
+
+  const handleChangeBannerImage = (callback: number) => {
+    setChosenBanner(callback);
+  };
+
+  const handleChangeBurger = (arg: boolean) => {
+    setIsBurgerOpen(arg);
+  };
+
+  const handleClickSwitchBurger = () => {
+    setIsBurgerOpen((prev) => !prev);
+  };
+
+  const sortList = (list: Product[], sortType: SortType) => {
+    return [...list].sort((a, b) => {
+      switch (sortType) {
+        case "Newest":
+          return b.year - a.year;
+        case "Alphabet":
+          return a.name.localeCompare(b.name);
+        case "Expensive":
+          return b.fullPrice - a.fullPrice;
+        case "Cheaper":
+          return a.fullPrice - b.fullPrice;
+      }
```

Comment:
sort is not working cos sortType has additional quotation marks. These extra quotes prevent sortType from matching any of the expected values. if you replace them by e.g sortType.replace(/^['"]|['"]$/g, ''), sorting will work properly ;)